src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -1043,14 +1077,26 @@ void GoogleCalendarSyncAdaptor::updateLocalCalendarNotebookEvents(int accountId,
 
             m_idDb.removeEvent(accountId, eventId);
             if (allMap.contains(eventId)) {
+                // currently existing base event or persistent occurrence which needs deletion
+                SOCIALD_LOG_DEBUG("Event deleted remotely:" << eventId);
                 m_calendar->deleteEvent(allMap.value(eventId));
                 m_storageNeedsSave = true;
+            } else if (allMap.contains(parentId)) {
+                // this is a non-persistent occurrence, we need to add an EXDATE to the base event.
+                SOCIALD_LOG_DEBUG("Occurrence deleted remotely:" << eventId << "for recurrenceId:" << recurrenceId.toString());
+                KCalCore::Event::Ptr event = allMap.value(parentId);
+                event->startUpdates();
+                event->recurrence()->addExDateTime(recurrenceId);
```

Comment:
Somehow not following here. So an exception which wasn't created locally, is removed remotely(?), google won't include an exception in recurrence rules after that?